File: Sem1/Assignment/code/LBEP_exam1.c
#include <stdio.h>
#include <string.h>
struct Book{
	char name[40];
	int quantity;
	int price;
};
void factoria(int* num){
	int num2 = *num;
	while(num2>1){
		num2--;
		(*num) *= num2;
	};
}

void perform(){
	printf("************************************\n");
	printf("C Program language\n");
	printf("\t1. Question 02.\n");
	printf("\t2. Question 03.\n");
	printf("\t3. Exit.\n");
	printf("************************************\n");
	printf("Enter Choice (1-3): ");
}
void question2(){
		int num2;
		printf("Enter an integer: ");
		scanf("%d",&num2);
		int number = num2;
		factoria(&num2);
		printf("Factoria: %d! = %d\n",number, num2);
}
void question3(){
	int num2,num3,num4;
	char a;
	double x,y;
	printf("How many books would you like to manage: ");
		scanf("%d",&num2);
		while(num2 > 10){
			printf("The number is invalid! Please any key to continue\n");
			scanf("%c",&a);
			printf("How many books would you like to manage: ");
			scanf("%d",&num2);
		};
		struct Book arr_book[num2];
		for(int i= 0; i < num2; i++){
			struct Book book;
			int num = i+1;
			char str[40];
			printf("Enter book %d name: ",num);
			scanf("%s",book.name);
			printf("Enter quantity: ");
			scanf("%d",&num3);
			book.quantity = num3;
			printf("Enter Unit price: ");
			scanf("%d",&num4);
			book.price = num4;
			printf("\n");
			arr_book[i] = book;
		};
		printf("\nBook List Information: \n");
		printf("No\tName\tQuantity Price Amount\n");
		printf("__|____________|_______|_______|______\n");
		for(int i = 0; i < num2;i++){
			int am = arr_book[i].quantity * arr_book[i].price;
			int j = i+1;
			printf("%d \t %s \t %d \t %d \t %d\n",j,arr_book[i].name,arr_book[i].quantity,arr_book[i].price,am);
		}
}
int main() {
	int num =1;
	while(num != 3){
	perform();
	scanf("%d",&num);
	if(num == 1){
		question2();
	} else if(num == 2){
		question3();
	}else if(num == 3){
		break;
	}else {
		printf("Nhập sai số\n Vui lòng chọn lại \n");
		perform();
	};
	}
}

File: Sem1/Assignment/code/LBEP_exam1.java
import java.util.*;

public class LBEP_exam1 {
	public static void main(String[] args) {
		System.out.println("************************************");
		System.out.println("Java Program language");
		System.out.println("\t1. Question 02.");
		System.out.println("\t2. Question 03.");
		System.out.println("\t3. Exit.");
		System.out.println("************************************");
		System.out.println("Enter Choice (1-3): ");
		Scanner input = new Scanner(System.in);
		String str = input.next();
		int num = Integer.parseInt(str);
		if (num == 1) {
			System.out.print("Enter an integer: ");
			String str2 = input.next();
			int num2 = Integer.parseInt(str2);
			int j = num2;
			int i = 1;
			while (num2 > 0) {
				i *= num2;
				num2--;
			}
			;
			System.out.printf("Factorial: %d! = %d\n", j, i);
		} else if (num == 2) {
			System.out.print("How many books would you like to manage: ");
			String str2 = input.next();
			int num_books = Integer.parseInt(str2);
			while (num_books > 5 || num_books <= 0) {
				System.out.println("The number is invalid! Please try again!\n");
				System.out.print("How many books would you like to manage: ");
				String num_books_str = input.next();
				num_books = Integer.parseInt(num_books_str);
			}
			;
			Book[] books = new Book[num_books];
			for (int i = 0; i < num_books; i++) {
				int num_i = i + 1;
				System.out.printf("Enter book %d name: ", num_i);
				String book_name = input.next();
				System.out.print("Enter quantity: ");
				String quan = input.next();
				int quan_num = Integer.parseInt(quan);
				System.out.print("Enter Unit price: ");
				String pri_str = input.next();
				int price = Integer.parseInt(pri_str);
				Book book = new Book(book_name,quan_num,price);
				books[i] = book;
				System.out.println();
			}
			;
			System.out.println("Book List Information: ");
			System.out.println("No\tName\tQuantity Price Amount");
			System.out.println("__|__________|_________|_____|______");
			for (int i = 0; i < num_books; i++) {
				int num_i = i + 1;
				System.out.printf("%d\t",num_i);
				books[i].GetInfo();
			}
			;

		}
	}
}
